packages/shorebird_cli/lib/src/validators/shorebird_yaml_asset_validator.dart

```diff
@@ -0,0 +1,108 @@
+import 'dart:io';
+import 'package:path/path.dart' as p;
+import 'package:shorebird_cli/src/shorebird_env.dart';
+import 'package:shorebird_cli/src/validators/validators.dart';
+import 'package:yaml/yaml.dart';
+import 'package:yaml_edit/yaml_edit.dart';
+
+class ShorebirdYAMLAssetValidator extends Validator {
+  final String pubspecYamlPath = p.join('pubspec.yaml');
+
+  @override
+  String get description => 'pubspec.yaml contains shorebird.yaml in assets';
+
+  @override
+  bool canRunInCurrentContext() => _pubspecYamlFile?.existsSync() ?? false;
+
+  @override
+  String get incorrectContextMessage => '''
+The ${_pubspecYamlFile?.path ?? 'pubspec.yaml'} file does not exist.
+The command you are running must be run within a Flutter app project.''';
+
+  @override
+  Future<List<ValidationIssue>> validate() async {
+    final pubspecYamlFile = _pubspecYamlFile;
+    if (pubspecYamlFile == null || !pubspecYamlFile.existsSync()) {
+      return [
+        ValidationIssue(
+          severity: ValidationIssueSeverity.error,
+          message: '''No pubspec.yaml file found at ${pubspecYamlFile?.path}''',
+        ),
+      ];
+    }
+
+    final pubspecContent = pubspecYamlFile.readAsStringSync();
+    final pubspecYaml = loadYaml(pubspecContent) as YamlMap;
+
+    if (!_pubspecYamlHasShorebirdAsset(pubspecYaml)) {
+      return [
+        ValidationIssue(
+          severity: ValidationIssueSeverity.error,
+          message: '$pubspecYamlPath is missing shorebird.yaml in assets',
+          fix: () => _addShorebirdAssetToFile(pubspecYamlFile),
+        ),
+      ];
+    }
+
+    return [];
+  }
+
+  File? get _pubspecYamlFile {
+    final root = shorebirdEnv.getFlutterProjectRoot();
+    if (root == null) return null;
+    return File(p.join(root.path, 'pubspec.yaml'));
+  }
+
+  bool _pubspecYamlHasShorebirdAsset(YamlMap pubspecYaml) {
+    final flutterSection = pubspecYaml['flutter'];
+    if (flutterSection is! YamlMap) return false;
+
+    final assetsSection = flutterSection['assets'];
+    if (assetsSection is! YamlList) return false;
+
+    return assetsSection.any((asset) => asset == 'shorebird.yaml');
+  }
+
+  void _addShorebirdAssetToFile(File pubspecYamlFile) {
```

Comment:
This looks like it's a duplicate of the `_addShorebirdYamlToProject` in `init_command.dart`. We should refactor so that the logic isn't duplicated.

Reply:
I found an even better function ` _addShorebirdYamlToPubspecAssets` from the `InitCommand` clsas. 

I made it static and public so that I can tap into that logic instead, it greatly reduced the code.

Is it ok to make that function static and public like that @felangel ?